### PR TITLE
Add 'disable_google_blockly' DCDO flag for transitioning labs

### PR DESCRIPTION
**Pull request by maddiedierker**
Allows us to override using Google Blockly for a level type using the `disable_google_blockly` DCDO flag. Example usage of this flag and how it should be set in the Rails console:

```ruby
[development] dashboard > DCDO.set('disable_google_blockly', ["Poetry"])
=> nil
[development] dashboard > DCDO.get('disable_google_blockly', [])
=> ["Poetry"]
```

Some rules about this flag:
1. It should be used for levels/labs that are transitioning to Google Blockly; it can be thought of as an emergency/test switch in case we need to switch back to CDO Blockly immediately because it will override `Level#uses_google_blockly?`.
2. The string added to this DCDO array should match the Level subclass name you want to override.
3. It is case insensitive (e.g., either "Poetry" or "poetry" will work in the example above).

Still supports the previous cases, which are:
1. fully migrated level types (e.g., all Flappy levels should use Google Blockly)
2. `view_options[:useGoogleBlockly]`, which right now is set when you visit a level/script_level with the `?blocklyVersion=Google` URL param

The migration path to transition a level type to Google Blockly (outside of any implementation/fixes needed on the client):
1. Implement `uses_google_blockly?` on the corresponding model class to return `true`. Once this is deployed, that level type will use Google Blockly by default.
2. Add the model class name to the list of class names in `DCDO.get('disable_google_blockly', [])` in the preferred environment if you need to immediately switch back to using CDO Blockly.

## Links

- [STAR-1896](https://codedotorg.atlassian.net/browse/STAR-1896)

## Testing story

Adds unit tests.

### CODE REVIEW

**Comment by ajpal** [comment]
Sorry maybe I'm missing something, but how will this work as an emergency flip? Wouldn't we need the inverse, like *don't* use google blockly, even if `level.uses_google_blockly?` returns `true`? Like, when we launch for Poetry, presumably we'll add a function like the one in Flappy, but then if we needed to quickly revert, we'd want to set the DCDO flag to force Poetry back to CdoBlockly

**Comment by maddiedierker** [comment]
> Like, when we launch for Poetry, presumably we'll add a function like the one in Flappy

i was assuming that we _won't_ add that function until we're fully ready to complete Poetry's migration to Google Blockly. so "Poetry" would stay in the DCDO configuration until we're ready to solidify it in the code. i assumed this because this is how we did the migration for Flappy, but maybe that's not actually what we want?

**Comment by maddiedierker** [comment]
that was also the conclusion i thought we came to in [this thread](https://codedotorg.slack.com/archives/C07TL2HV2/p1636491705006600)

**Comment by ajpal** [comment]
ahh got it, I misread the thread then!

**Review by ajpal — APPROVED** [review]

**Comment by ajpal** [comment]
ah wait, sorry for the back and forth, but I think we actually do want the inverse logic. from @breville in the slack thread
> fwiw, I think it's reasonable to move tutorials to new blockly via code, and just have an optional dcdo flag as a way to flip back to old blockly in case of emergency.  i.e. use code as the source of truth, and dcdo only as an emergency override, rather than the source of truth living in dcdo.  it means that most of the time, the dcdo flag remains empty.

So then we'd want to start out with the dcdo flag empty, and only use it if something happens and we need to revert?

**Comment by maddiedierker** [comment]
ah, okay -- my last message in that thread says the opposite, so i must've misread.

what matters is the behavior we _do_ want because either way will work fine. so summarizing what i think we want:
- implement `Poetry#uses_google_blockly?` to return true
- update the DCDO flag to `disable_google_blockly` and it will override the method above
- always allow `?blocklyVersion=Google` to override either

**Comment by ajpal** [comment]
> * implement `Poetry#uses_google_blockly?` to return true
> * update the DCDO flag to `disable_google_blockly` and it will override the method above
> * always allow `?blocklyVersion=Google` to override either

Yes, this is my understanding of correct behavior as well. Sorry for the churn here!

File: dashboard/test/helpers/levels_helper_test.rb [hunk 1]
```diff
@@ -247,6 +247,42 @@ def request
...
+
+    Level.unstub(:uses_google_blockly?)
+  end
+
+  test 'use_google_blockly is false if level.uses_google_blockly? but disable_google_blockly is set' do
+    GamelabJr.any_instance.stubs(:uses_google_blockly?).returns(true)
+    @level = build :spritelab
+    assert use_google_blockly
+
+    DCDO.stubs(:get).with('disable_google_blockly', []).returns(['GamelabJr'])
```
Comment (breville) [inline dashboard/test/helpers/levels_helper_test.rb:276]: I think the DCDO array values can be all lowercase?  (Similar comment for example in the PR description.)

Comment (maddiedierker) [reply dashboard/test/helpers/levels_helper_test.rb:276]: they can -- i added this case and the one after it to display that these DCDO values are case insensitive, but i think it's clearer to display that the strings in this array should match the Level subclass name, so i'd like to leave the casing consistent even though either way will work. i'll add a note to the PR description about it though

File: dashboard/app/models/levels/poetry.rb [hunk 2]
```diff
@@ -73,6 +73,10 @@ def self.create_from_level_builder(params, level_params)
     )
   end
 
+  def uses_google_blockly?
+    true
```
Comment (ajpal) [inline dashboard/app/models/levels/poetry.rb:77]: 🤞

**Review by ajpal — APPROVED** [review]

File: dashboard/app/models/levels/poetry.rb (hunk 2 above)
Comment (maddiedierker) [reply dashboard/app/models/levels/poetry.rb:77]: i have high hopes! should i ship this PR today (and, optionally, disable Google Blockly in whatever environments we want before this change hits them), or would you rather it ship tomorrow/later this week?

Comment (ajpal) [reply dashboard/app/models/levels/poetry.rb:77]: If you don't mind, can you wait until tomorrow or friday when the location picker bug is fixed?

Comment (maddiedierker) [reply dashboard/app/models/levels/poetry.rb:77]: yeah, of course!

**Pull request merged**